hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/WorkloadStat.java

```diff
@@ -33,9 +33,9 @@ public class WorkloadStat implements Serializable {
 
   private long numUpdates = 0L;
 
-  private HashMap<String, Pair<String, Long>> insertLocationToCount;
+  private final HashMap<String, Pair<String, Long>> insertLocationToCount;
```

Comment:
Also, for Java serialization and deserialization to work in Spark, these member variables cannot be final.

Reply:
I think for any class that already has final variables, it is safe to add additional `final` variables.  Otherwise, we need to be careful around the `Serializable`.